### PR TITLE
Fix `gem exec rails new project` failing on Ruby 3.2

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
## What was the end-user or developer problem that led to this PR?

On Ruby 3.2, when running `gem exec rails new project`, the default version of securerandom (0.2.2) gets activated, but does not match Rails requirements (>= 0.3), leading to an error like this:

```
$ gem exec rails new repro
/Users/deivid/Code/rubygems/rubygems/lib/rubygems/specification.rb:2246:in `raise_if_conflicts': Unable to activate activesupport-7.2.1, because securerandom-0.2.2 conflicts with securerandom (>= 0.3) (Gem::ConflictError)
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/specification.rb:1383:in `activate'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/specification.rb:1421:in `block in activate_dependencies'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/specification.rb:1403:in `each'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/specification.rb:1403:in `activate_dependencies'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/specification.rb:1385:in `activate'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/core_ext/kernel_gem.rb:62:in `block in gem'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/core_ext/kernel_gem.rb:62:in `synchronize'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/core_ext/kernel_gem.rb:62:in `gem'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/commands/exec_command.rb:193:in `activate!'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/commands/exec_command.rb:73:in `execute'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/command.rb:326:in `invoke_with_build_args'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/command_manager.rb:255:in `invoke_command'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/command_manager.rb:194:in `process_args'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/command_manager.rb:152:in `run'
	from /Users/deivid/Code/rubygems/rubygems/lib/rubygems/gem_runner.rb:56:in `run'
	from /Users/deivid/code/rubygems/rubygems/exe/gem:12:in `<main>'
```

## What is your fix for the problem, implemented in this PR?

Vendor securerandom.

## Make sure the following tasks are checked

- [x] Describe the problem / feature
- [ ] Write [tests](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#tests) for features and bug fixes
- [x] Write code to solve the problem
- [x] Make sure you follow the [current code style](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#code-formatting) and [write meaningful commit messages without tags](https://github.com/rubygems/rubygems/blob/master/bundler/doc/development/PULL_REQUESTS.md#commit-messages)

### CODE REVIEW

**Review by indirect — APPROVED** [review]

**Comment by casperisfine** [comment]
❤️

File: lib/rubygems/vendor/securerandom/lib/random/formatter.rb [hunk 1]
```diff
@@ -0,0 +1,373 @@
...
+#
+#   prng.uuid #=> "2d931510-d99f-494a-8c67-87feb05e1594"
+#   prng.uuid #=> "bad85eb9-0713-4da7-8d36-07a8e4b00eab"
+#   Random.uuid #=> "f14e0271-de96-45cc-8911-8910292a42cd"
+#
+# All methods are available in the standard library Gem::SecureRandom, too:
+#
+#   Gem::SecureRandom.hex #=> "05b45376a30c67238eb93b16499e50cf"
+
+module Random::Formatter
```
Comment (byroot) [inline lib/rubygems/vendor/securerandom/lib/random/formatter.rb:55]: Just noticed this wasn't namespaced. I think that will cause redefinition warnings.

Comment (deivid-rodriguez) [reply lib/rubygems/vendor/securerandom/lib/random/formatter.rb:55]: Good catch! I think we'll need to tweak automatiek to handle this.

Comment (byroot) [reply lib/rubygems/vendor/securerandom/lib/random/formatter.rb:55]: Yeah, secure random is weird. Upstream is aware but unclear what they'll do: https://github.com/ruby/securerandom/pull/29

**Pull request merged**